src/ILCompiler.Compiler/src/Compiler/InteropStubManager.cs

```diff
@@ -0,0 +1,98 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.IO;
+using System.Collections.Generic;
+
+using Internal.IL.Stubs;
+using Internal.TypeSystem;
+using Internal.TypeSystem.Interop;
+
+using ILCompiler.Metadata;
+using ILCompiler.DependencyAnalysis;
+using ILCompiler.DependencyAnalysisFramework;
+
+using Debug = System.Diagnostics.Debug;
+using ReadyToRunSectionType = Internal.Runtime.ReadyToRunSectionType;
+using ReflectionMapBlob = Internal.Runtime.ReflectionMapBlob;
+
+namespace ILCompiler
+{
+    /// <summary>
+    /// This class is responsible for managing stub methods for interop
+    /// </summary>
+    public sealed class InteropStubManager
```

Comment:
This seems to be doing very similar things with InteropStateManager. What does the additional layers bring to the table?